pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +87,29 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func WaitAndRetryForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	rand.NewSource(time.Now().UnixNano()) //creates a more unique random
```

Comment:
Why isn't a permanent solution?  These retry-able errors are normal in distributed systems.   

Also I'm not getting why rand is better than expo backoff.